pkg/providers/tinkerbell/validator.go

```diff
@@ -0,0 +1,152 @@
+		return errors.New("must specify machineGroupRef for worker nodes")
+	}
+
+	workerNodeGroupMachineConfig := tinkerbellClusterSpec.firstWorkerMachineConfig()
+	if workerNodeGroupMachineConfig == nil {
+		return fmt.Errorf("cannot find tinkerbellMachineConfig %v for worker nodes", tinkerbellClusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name)
+	}
+	if len(workerNodeGroupMachineConfig.Spec.OSFamily) <= 0 {
+		return errors.New("tinkerbellMachineConfig osFamily for worker nodes is not set or is empty")
+	}
+
+	// TODO: move this to api Cluster validations
+	controlPlaneEndpointIp := tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host
+	if controlPlaneEndpointIp == "" {
+		return fmt.Errorf("controlPlaneConfiguration.Endpoint.Host is required")
+	}
+	if err := v.validateIP(tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host); err != nil {
+		return fmt.Errorf("cluster controlPlaneConfiguration.Endpoint.Host is invalid: %s", controlPlaneEndpointIp)
+	}
+
+	if controlPlaneMachineConfig.Spec.OSFamily != anywherev1.Bottlerocket && controlPlaneMachineConfig.Spec.OSFamily != anywherev1.Ubuntu {
+		return fmt.Errorf("control plane osFamily: %s is not supported, please use one of the following: %s, %s", controlPlaneMachineConfig.Spec.OSFamily, anywherev1.Bottlerocket, anywherev1.Ubuntu)
+	}
+
+	if workerNodeGroupMachineConfig.Spec.OSFamily != anywherev1.Bottlerocket && workerNodeGroupMachineConfig.Spec.OSFamily != anywherev1.Ubuntu {
+		return fmt.Errorf("worker node osFamily: %s is not supported, please use one of the following: %s, %s", workerNodeGroupMachineConfig.Spec.OSFamily, anywherev1.Bottlerocket, anywherev1.Ubuntu)
+	}
+
+	if controlPlaneMachineConfig.Spec.OSFamily != workerNodeGroupMachineConfig.Spec.OSFamily {
+		return errors.New("control plane and worker nodes must have the same osFamily specified")
+	}
+
+	for _, machineConfig := range tinkerbellClusterSpec.machineConfigsLookup {
+		if machineConfig.Namespace != tinkerbellClusterSpec.Cluster.Namespace {
+			return errors.New("TinkerbellMachineConfig and Cluster objects must have the same namespace specified")
+		}
+	}
+
+	if tinkerbellClusterSpec.datacenterConfig.Namespace != tinkerbellClusterSpec.Cluster.Namespace {
+		return errors.New("TinkerbellDatacenterConfig and Cluster objects must have the same namespace specified")
+	}
+
+	return nil
+}
+
+func (v *Validator) validateTinkerbellIP(ctx context.Context, ip string) error {
+	// check if tinkerbellIP is valid
+	if ip == "" {
+		return fmt.Errorf("tinkerbellIP is required")
+	}
+
+	if err := v.validateIP(ip); err != nil {
+		return fmt.Errorf("cluster tinkerbellDatacenterConfig.tinkerbellIP is invalid: %s", ip)
+	}
+	return nil
+}
+
+func (v *Validator) validateIP(ip string) error {
+	parsedIp := net.ParseIP(ip)
+	if parsedIp == nil {
+		return fmt.Errorf("IP is invalid: %s", ip)
+	}
+
+	return nil
```

Comment:
Since we seem to do IP validation in a lot of places, can we just move this section to the netutils package? https://github.com/aws/eks-anywhere/blob/main/pkg/networkutils/networkutils.go